routers/web/repo/issue.go

```diff
@@ -1242,33 +1242,61 @@ func roleDescriptor(ctx stdCtx.Context, repo *repo_model.Repository, poster *use
 	// By default the poster has no roles on the comment.
 	roleDescriptor := issues_model.RoleDescriptorNone
 
+	// If the poster is the actual poster of the issue, enable Poster role.
+	if issue.IsPoster(poster.ID) {
+		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorPoster)
+	}
+
 	// Check if the poster is owner of the repo.
 	if perm.IsOwner() {
 		// If the poster isn't a admin, enable the owner role.
 		if !poster.IsAdmin {
 			roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorOwner)
-		} else {
+			return roleDescriptor, nil
+		}
 
-			// Otherwise check if poster is the real repo admin.
-			ok, err := access_model.IsUserRealRepoAdmin(repo, poster)
-			if err != nil {
-				return issues_model.RoleDescriptorNone, err
-			}
-			if ok {
-				roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorOwner)
-			}
+		// Otherwise check if poster is the real repo admin.
+		ok, err := access_model.IsUserRealRepoAdmin(repo, poster)
+		if err != nil {
+			return issues_model.RoleDescriptorNone, err
+		}
+		if ok {
+			roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorOwner)
+			return roleDescriptor, nil
 		}
 	}
 
-	// Is the poster can write issues or pulls to the repo, enable the Writer role.
-	// Only enable this if the poster doesn't have the owner role already.
-	if !roleDescriptor.HasRole("Owner") && perm.CanWriteIssuesOrPulls(issue.IsPull) {
-		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorWriter)
+	// If repo is organization, check Member role
+	if err := repo.LoadOwner(ctx); err != nil {
+		return issues_model.RoleDescriptorNone, err
+	}
+	if repo.Owner.IsOrganization() {
+		if isMember, err := organization.IsOrganizationMember(ctx, repo.Owner.ID, poster.ID); err != nil {
+			return issues_model.RoleDescriptorNone, err
+		} else if isMember {
+			roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorMember)
+			return roleDescriptor, nil
+		}
 	}
 
-	// If the poster is the actual poster of the issue, enable Poster role.
-	if issue.IsPoster(poster.ID) {
-		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorPoster)
+	// If the poster is the collaborator of the repo
+	if isCollaborator, err := repo_model.IsCollaborator(ctx, repo.ID, poster.ID); err != nil {
+		return issues_model.RoleDescriptorNone, err
+	} else if isCollaborator {
+		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorCollaborator)
+		return roleDescriptor, nil
+	}
+
+	if commits, err := gitRepo.SearchCommits(git.SearchCommitsOptions{
+		Authors: []string{poster.Name},
+		All:     true,
+		Limit:   2,
+	}); err != nil {
+		return issues_model.RoleDescriptorNone, err
+	} else if len(commits) == 1 {
+		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorContributor)
+	} else if len(commits) > 1 {
+		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorFirstTimeContributor)
```

Comment:
Shouldn't it be the other way around?
```suggestion
	} else if len(commits) > 1 {
		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorContributor)
	} else if len(commits) == 1 {
		roleDescriptor = roleDescriptor.WithRole(issues_model.RoleDescriptorFirstTimeContributor)
```

Reply:
Done.